app/controllers/response_controller.rb

```diff
@@ -200,49 +205,28 @@ def pending_surveys
       return
     end
 
-    # Get all the participant(course or assignment) entries for this user
-    course_participants = CourseParticipant.where(user_id: session[:user].id)
-    assignment_participants = AssignmentParticipant.where(user_id: session[:user].id)
-
     # Get all the course survey deployments for this user
     @surveys = []
-    if course_participants
-      course_participants.each do |cp|
-        survey_deployments = CourseSurveyDeployment.where(parent_id: cp.parent_id)
-        next unless survey_deployments
-        survey_deployments.each do |survey_deployment|
-          next unless survey_deployment && Time.now > survey_deployment.start_date && Time.now < survey_deployment.end_date
-          @surveys <<
-          [
-            'survey' => Questionnaire.find(survey_deployment.questionnaire_id),
-            'survey_deployment_id' => survey_deployment.id,
-            'start_date' => survey_deployment.start_date,
-            'end_date' => survey_deployment.end_date,
-            'parent_id' => cp.parent_id,
-            'participant_id' => cp.id,
-            'global_survey_id' => survey_deployment.global_survey_id
-          ]
-        end
-      end
-    end
-
-    # Get all the assignment survey deployments for this user
-    if assignment_participants
-      assignment_participants.each do |ap|
-        survey_deployments = AssignmentSurveyDeployment.where(parent_id: ap.parent_id)
+    [CourseParticipant, AssignmentParticipant].each do |item|
+      # Get all the participant(course or assignment) entries for this user
+      participant_type = item.where(user_id: session[:user].id)
+      next unless participant_type
+      participant_type.each do |p|
+        survey_deployment_type = participant_type == CourseParticipant ? AssignmentSurveyDeployment : CourseSurveyDeployment
```

Comment:
I don't think the logic here is correct. `participant_type` is an array of participants, instead of the type of participant. I think this mistake is due to your misleading variable names. And you just refactor code without varifying the correctness.